.github/workflows/ci-tests.yml

```diff
@@ -32,40 +32,49 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os:
-          - ubuntu-22.04
-          - macos-12
-          - windows-2022
+        os: ${{ fromJson(github.repository_owner == 'subspace' && github.repository == 'subspace/subspace-cli' && '[["self-hosted", "ubuntu-22.04-x86-64", "Linux"], ["self-hosted", "macos-12-arm64", "macOS"], ["self-hosted", "windows-server-2022-x86-64", "Windows"]]' || '["ubuntu-22.04", "macos-12", "windows-2022"]') }}
         run-all:
           - ${{ inputs.test-macos-and-windows == true || github.ref == 'refs/heads/main' }}
         exclude: # exclude macos-12 and windows-2022 when the condition is false
           - run-all: false
             os: macos-12
           - run-all: false
             os: windows-2022
+          - run-all: false
+            os: macOS
+          - run-all: false
+            os: Windows
 
     runs-on: ${{ matrix.os }}
     steps:
       - name: git checkout
         uses: actions/checkout@93ea575cb5d8a053eaa0ac8fa3b40d7e05a33cc8 # @v3.1.0
 
-      # On macOS, we need a proper Clang version, not Apple's custom version without wasm32 support
+      # On macOS, we need a proper Clang version, not Apple's custom version without wasm32 support, we also install clang on Windows
```

Comment:
Shouldn't need to install on Windows on every run, it must be present right away just like it is with GitHub's runners.

Reply:
I found a fix for this in visual studio developer tools to install llvm,clang so we don't need this anymore. Although i installed  https://visualstudio.microsoft.com/vs/features/cplusplus/ and build tools, it seems like clang is another package https://learn.microsoft.com/en-us/cpp/build/clang-support-msbuild?view=msvc-170
will test this out.

Reply:
I'm not sure what MSVC has to do with LLVM, I'd expect you to install it from here: https://github.com/llvm/llvm-project/releases/tag/llvmorg-15.0.7 (latest release of 15.y.z series). We probably don't want some random version of LLVM and Clang installed.

Reply:
It's the same one bundled in visual studio C++ build tools, not a "random" one.

Reply:
so why don't we just install it rather with the runner rather than using the windows implementation,  installing and building from source is more a nuisance to maintain constantly.

Reply:
Honestly, I do not understand what you're talking about. MSVC and Clang are different compilers, we need both installed on the host the same way they are on GitHub's runners (but, obviously, when installing MSVC we'll be getting community version, not whatever paid version GitHub uses).